src/sql/magic.py

```diff
@@ -147,6 +147,14 @@ class SqlMagic(Magics, Configurable):
         config=True,
         help="Verbosity level. 0=minimal, 1=normal, 2=all",
     )
+    lazy_spark = Bool(
```

Comment:
Might rename this config to lazy_execution, then it could be used to close #536  following a similar pattern in the run_statement function